### PR TITLE
Slightly better error messages

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Fixes https://github.com/googleapis/nodejs-firestore/issues/436 and https://b.corp.google.com/issues/78932093

This PR:
- adds the field path to the error message if a field-level validation fails
- uses consistent quotes in all errors (changed single quotes to double)
- enforces that all errors are verified as they are raised by removing the RegExp matching and matching by exact value instead

### CODE REVIEW

File: dev/src/index.ts [hunk 1]
```diff
@@ -289,8 +289,9 @@ export class Firestore {
    */
   constructor(settings?: Settings) {
     this._validator = new Validator({
-      ArrayElement: (name, value) =>
-          validateFieldValue(name, value, /* depth */ 0, /*inArray=*/true),
+      ArrayElement: (name, value) => validateFieldValue(
+          name, value, /* path */ undefined, /*level=*/0,
```
Comment (mikelehen) [inline dev/src/index.ts]: super nit: path=

File: dev/src/index.ts [hunk 2]
```diff
@@ -1261,51 +1262,67 @@ follow these steps, YOUR APP MAY BREAK.`);
  *
  * @private
  * @param val JavaScript value to validate.
+ * @param path The field path to validate.
  * @param options Validation options
- * @param depth The current depth of the traversal.
+ * @param level The current depth of the traversal. This is used to decided
```
Comment (mikelehen) [inline dev/src/index.ts]: decided=>decide

File: dev/src/path.ts [hunk 3]
```diff
@@ -477,20 +481,29 @@ export class FieldPath extends Path<FieldPath> {
    */
   static validateFieldPath(fieldPath: string|FieldPath) {
     if (!(fieldPath instanceof FieldPath)) {
-      if (!is.string(fieldPath)) {
+      if (fieldPath === undefined) {
```
Comment (mikelehen) [inline dev/src/path.ts]: FWIW this seems to contradict the type signature of the method, which is weird / confusing.  Perhaps we should be handling undefined higher up, or we should fix the signature to indicate that we don't actually know what fieldPath is?

File: dev/src/index.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply dev/src/index.ts]: Fixed

File: dev/src/index.ts (hunk 2 above)
Comment (schmidt-sebastian) [reply dev/src/index.ts]: Fixed

**Pull request merged**